modules/ding_gatewayf/ding_gatewayf.module

```diff
@@ -241,6 +246,70 @@ function ding_gatewayf_callback() {
 }
 
 /**
+ * Do a check on the checksum given from gatewayf.
+ *
+ * @param array $attributes
+ */
+function ding_gatewayf_security_check($attributes) {
+  //Check attributes.
+  if (empty($attributes['schacPersonalUniqueID']) || empty($attributes['eduPersonTargetedID'])) {
+    // We expect alway to get these two attributes.
+    watchdog('ding_gatewayf', 'Neccesary attributes missing', array(), WATCHDOG_ERROR);
+    drupal_set_message('Sorry login failed - contact site administrator', 'warning');
+    ding_gatewayf_user_logout();
+  }
+
+  $gateway_checksum = isset($attributes['hash'][0]) ? $attributes['hash'][0] : NULL;
+  if (!isset($gateway_checksum)) {
+    //This is an error from gatewayf - checksum should always be in attributes.
+    watchdog('ding_gatewayf', 'No checksum from gatewayf', array(), WATCHDOG_ERROR);
+    drupal_set_message('Sorry login failed - contact site administrator', 'warning');
+    ding_gatewayf_user_logout();
+  }
+
+  $fixed_params = array(
+    $attributes['schacPersonalUniqueID'][0],
+    $attributes['eduPersonTargetedID'][0],
+  );
+  $key = ding_gatewayf_key_from_attributes($fixed_params);
+  //Get salt from settings $conf.
+  $salt = variable_get('ding_gatewayf_salt', NULL);
+  if (empty($salt)) {
+    //This is a configuration error.
+    watchdog('ding_gatewayf', 'No salt has been defined - wayf login is disabled for security', array(), WATCHDOG_ERROR);
+    drupal_set_message('No salt has been defined - wayf login is disabled for security');
+    ding_gatewayf_user_logout();
```

Comment:
Why do we want to log out the user here?

I would think that the assumption is that if the security check fails then the user has not actually been logged in through gatewayf anyway so there is no reason to log out either.